dlib/image_loader/png_loader.cpp

```diff
@@ -190,25 +260,25 @@ namespace dlib
             color_type_ != PNG_COLOR_TYPE_RGB_ALPHA &&
             color_type_ != PNG_COLOR_TYPE_GRAY_ALPHA)
         {
-            fclose( fp );
+            if ( fp != NULL ) fclose( fp );
             png_destroy_read_struct( &( ld_->png_ptr_ ), &( ld_->info_ptr_ ), &( ld_->end_info_ ) );
-            throw image_load_error(std::string("png_loader: unsupported color type in file ") + filename);
+            throw image_load_error(std::string("png_loader: unsupported color type"));
```

Comment:
Make this something like 
```
throw image_load_error("png_loader: unsupported color type" + file_info);
```
and put `" in file " + filename` in `file_info` when we know the filename.  Otherwise the error is going to be fairly confusing to users if they encounter it.